smtp_url: password cannot contain a question mark

## bug report

* Expected behavior
Neomutt says the smtp_url is incorrect when sending an e-mail.
* Actual behavior
The url is correct.
* Steps to reproduce
Set in `.muttrc`:

```set smtp_url=smtps://"aaa":"AA?"@server```

Nothing else is needed, this is the whole config file causing the problem.

* Used program versions
Bisected, the first bad commit is https://github.com/neomutt/neomutt/commit/4ac1c81ed710c8b27dc23821330b2005f7122214
* Operating System and its version
Archlinux, x86_64

Wow, that was fast. I did `./configure && make` and it fails with:

```
url.c: In function ‘url_parse’:
url.c:178:21: error: ‘U_NOTMUCH’ undeclared (first use in this function)
   if ((u->scheme == U_NOTMUCH) || (u->scheme == U_MAILTO))
```

Running `./configure --notmuch && make` works and solves the problem. I cannot confirm if it really sends the e-mail, because not everything needed is included in the built (for instance SSL support). How is neomutt built for production? Simply `./configure --everything`?

Also, when I run `neomutt "mailto:aa"` I get SIGSEV. However, it might be just some missing dependency??
Anyway, output of GDB:

```
GNU gdb (GDB) 8.0.1
(gdb) run "mailto:A"
Starting program: /tmp/neomutt/neomutt "mailto:A"
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Thread 1 "neomutt" received signal SIGSEGV, Segmentation fault.
mutt_rfc2047_decode (pd=pd@entry=0x555555950d20) at mutt/rfc2047.c:678
678	  while (*s)
(gdb) backtrace
#0  0x0000555555605930 in mutt_rfc2047_decode (pd=pd@entry=0x555555950d20) at mutt/rfc2047.c:678
#1  0x00005555555f5fe8 in url_parse_mailto (e=0x555555950ca0, body=0x7fffffffdd20, src=<optimized out>)
    at url.c:435
#2  0x000055555556d376 in main (argc=2, argv=0x7fffffffe538, env=<optimized out>) at main.c:568
```

--- a/mutt/rfc2047.c
+++ b/mutt/rfc2047.c
@@ -658,6 +658,9 @@ void mutt_rfc2047_encode(char **pd, const char *specials, int col, const char *c
  */
 void mutt_rfc2047_decode(char **pd)
 {
+  if (!pd || !*pd)
+    return;
+
   struct Buffer buf = { 0 }; /* Output buffer                          */
   char *s = *pd;             /* Read pointer                           */
   char *beg;                 /* Begin of encoded word                  */
@@ -745,5 +748,4 @@ void mutt_rfc2047_decode(char **pd)
 
   mutt_buffer_addch(&buf, '\0');
   *pd = buf.data;
-  return;
 }

--- a/url.c
+++ b/url.c
@@ -174,12 +174,20 @@ int url_parse(struct Url *u, char *src)
 
   src += 2;
 
-  t = strchr(src, '?');
-  if (t)
+  /* Notmuch and mailto schemes can include a query */
+#ifdef USE_NOTMUCH
+  if ((u->scheme == U_NOTMUCH) || (u->scheme == U_MAILTO))
+#else
+  if (u->scheme == U_MAILTO)
+#endif
   {
-    *t++ = '\0';
-    if (parse_query_string(u, t) < 0)
-      goto err;
+    t = strrchr(src, '?');
+    if (t)
+    {
+      *t++ = '\0';
+      if (parse_query_string(u, t) < 0)
+        goto err;
+    }
   }
 
   u->path = strchr(src, '/');

